relay-server/src/actors/project.rs

```diff
@@ -773,33 +758,32 @@ impl Handler<GetCachedProjectState> for Project {
 /// state is always refreshed.
 #[derive(Debug)]
 pub struct GetProjectState {
-    no_cache: bool,
+    pub public_key: ProjectKey,
+    pub no_cache: bool,
 }
 
 impl GetProjectState {
     /// Fetches the project state and uses the cached version if up-to-date.
-    pub fn new() -> Self {
-        Self { no_cache: false }
+    pub fn new(public_key: ProjectKey) -> Self {
+        Self {
+            public_key,
+            no_cache: false,
+        }
     }
 
     /// Fetches the project state and conditionally skips the cache.
-    pub fn no_cache(no_cache: bool) -> Self {
-        Self { no_cache }
+    pub fn no_cache(public_key: ProjectKey, no_cache: bool) -> Self {
```

Comment:
_nit_: This would now be better implemented as a builder:

```suggestion
    pub fn no_cache(mut self, no_cache: bool) -> Self {
```

You can then use it like this:

```rust
let message = GetProjectState::new(my_key).no_cache(true);
```

Reply:
done